src/LibraryManager.Vsix/UI/Controls/Library.xaml.cs

```diff
@@ -114,23 +153,30 @@ private void HandleKeyPress(object sender, KeyEventArgs e)
             }
         }
 
-        private void HandleKeyUp(object sender, KeyEventArgs e)
-        {
-            CaretIndex = LibrarySearchBox.CaretIndex;
-            RefreshSearch();
-        }
-
         private void HandleListBoxKeyPress(object sender, KeyEventArgs e)
         {
             int index = LibrarySearchBox.CaretIndex;
 
             switch (e.Key)
             {
-                case Key.Tab:
-                case Key.Enter:
+                 case Key.Tab:
+                    // SelectedItem could be null if the key press came from keyboard navigation and not commit operation.
+                    // In this case we will just move the focus to next control.
+                    if (SelectedItem == null)
+                    {
+                        LibrarySearchBox.MoveFocus(new TraversalRequest(FocusNavigationDirection.Next));
+                        e.Handled = true;
+                        break;
+                    }
+
                     Commit(SelectedItem);
+                    LibrarySearchBox.Focus();
                     e.Handled = true;
+                    break;
```

Comment:
Removed this as this should behave the same as Enter. I didn't realize this is for listbox key press. Selected item will never be null.